root/static/scripts/edit/URLCleanup.js

```diff
@@ -470,14 +535,32 @@ const CLEANUPS = {
     ],
     type: LINK_TYPES.lyrics,
     clean: function (url) {
-      url = url.replace(/^https:\/\/([a-z-]+\.)?wikisource\.org/, "http://$1wikisource.org");
       url = url.replace(/^https?:\/\/(.+\.)?genius\.com/, "http://$1genius.com");
       return url;
     }
   },
   bbcmusic: {
     match: [new RegExp("^(https?://)?(www\\.)?bbc\\.co\\.uk/music/artists/", "i")],
-    type: LINK_TYPES.bbcmusic
+    type: LINK_TYPES.bbcmusic,
+    clean: function (url) {
+      url = url.replace(/\/music\/artists\/([0-9a-f-]+)/, "http://www.bbc.co.uk/music/artists/$1");
+      return url;
+    },
+    validate: function (url, id) {
+      return /^http:\/\/www\.bbc\.co\.uk\/music\/artists\/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/.test(url);
```

Comment:
This could be tighten by adding the condition `_.includes(CLEANUPS.bbcmusic.type, id)` so to discard BBC Music artist pages for other link types, see `amazon` and `youtube` for examples.